### DISCUSSION TITLE
external-signing-and-pki

### DISCUSSION THREAD

**Discussion by kavithabk** [rauc/rauc#1534] [Q&A]
Hi all ,
We are using verity format for our RAUC bundle.
We are signing our RAUC manifest using External PKI.
We get a detached signature for the manifest.raucm from the external PKI.

How can we bundle manifest.raucm + detached signature + Public Key certificates together to an CMS envelope? so that RAUC will be able to  work new signature?Is there any suggestions on the same?

Ref: https://rauc.readthedocs.io/en/v1.12/reference.html#external-signing-and-pki

**Comment by jluebbe** [comment]
There is some background in https://github.com/rauc/rauc/issues/1532.

Please attach the extracted and new signature files. Also the PKI chain certificates would be useful.

**Comment by kavithabk** [comment]
openssl cms -verify -in attached_signature.cms -inform DER -noverify --print

CMS Verification successful

The verification is successful now.


But I get below error
openssl cms -verify -CAfile ../signer_service_RootCA_ECC.pem -inform der -signer out.cms -in attached_signature.cms 
CMS Verification failure
40876671157B0000:error:17000064:CMS routines:cms_signerinfo_verify_cert:certificate verify error:../crypto/cms/cms_smime.c:289:Verify error: unable to get local issuer certificate


I have attached the cms file for reference with some data removed
[log.txt](https://github.com/user-attachments/files/17375240/log.txt)

**Comment by kavithabk** [comment]
RooT CA was the issue will mark this working
